DigitalLearningSolutions.Web/Views/Login/Index.cshtml

```diff
@@ -1,12 +1,79 @@
-﻿@{
-  ViewData["Title"] = "Login";
+﻿@using DigitalLearningSolutions.Web.ViewModels.Login
+@model LoginViewModel 
+@{
+    var errorHasOccurred = !ViewData.ModelState.IsValid;
+    ViewData["Title"] = errorHasOccurred ? "Error: Log in" : "Log in";
+    var usernameError = ViewData.ModelState["Username"]?.Errors?.Count > 0;
+    var passwordError = ViewData.ModelState["Password"]?.Errors?.Count > 0;
+    var passwordInputErrorClass = passwordError ? "nhsuk-input--error" : "";
+    var passwordFormErrorClass = passwordError ? "nhsuk-form-group--error" : "";
+    var usernameInputErrorClass = usernameError ? "nhsuk-input--error" : "";
+    var usernameFormGroupError = usernameError ? "nhsuk-form-group--error" : "";
 }
 
 <div class="nhsuk-grid-row">
   <div class="nhsuk-grid-column-full">
-    <h1 id="page-heading">Login</h1>
-    <a class="nhsuk-button" asp-controller="Login" asp-action="SignIn">
-      Sign in
+    <form class="nhsuk-u-margin-bottom-8" method="post" novalidate asp-action="Index">
+
+      @if (errorHasOccurred) {
+        <partial name="_ErrorSummary"/>
+      }
+      <div class="nhsuk-u-margin-bottom-8">
+        <h1 id="page-heading">Log in</h1>
+        <p class="nhsuk-body-l">Enter your details to log into your account.</p>
+      </div>
+
+      <div class="nhsuk-form-group @usernameFormGroupError">
+        <label class="nhsuk-label" for="Username">
+          Email or user ID
+        </label>
+        @if (usernameError) {
+          <span class="nhsuk-error-message" id="username-error">
+            <span class="nhsuk-u-visually-hidden">Error:</span> @ViewData.ModelState["Username"].Errors[0].ErrorMessage
+          </span>
+        }
+        <input class="nhsuk-input nhsuk-u-width-one-half @usernameInputErrorClass" type="text" id="Username" spellcheck="false" autocomplete="email" aria-describedby="username-error" asp-for="Username" />
+      </div>
+
+      <div class="nhsuk-form-group @passwordFormErrorClass">
+        <label class="nhsuk-label" for="Password">
+          Password
+        </label>
+        @if (passwordError) {
+      <span class="nhsuk-error-message" id="password-error">
```

Comment:
The formatting here is a little bit odd - could you run the formatter on this? 🙏